spec/models/spree/gateway/stripe_sca_spec.rb

```diff
@@ -74,6 +75,46 @@
     end
   end
 
+  describe "#void", :vcr, :stripe_version do
+    context "with a confirmed payment" do
+      pending "refunds the payment" do
```

Comment:
@filipefurtad0 I am trying to test Stripe refund here but I am a bit blocked 
The `void` method retrieve the payment intent here : https://github.com/openfoodfoundation/openfoodnetwork/blob/4dda4e2385fcf45b95f4c2bfe745b2a98b7d42af/app/models/spree/gateway/stripe_sca.rb#L90-L91
It requires a `stripe_account_id`. The problem here is I don't know which `stripe_account_id` to use, I am using the Stripe test account you setup. Currently I get the following error :
```
 1) Spree::Gateway::StripeSCA#void with a confirmed payment refunds the payment
     Failure/Error:
       payment_intent_response = Stripe::PaymentIntent.retrieve(payment_intent_id,
                                                                stripe_account: stripe_account_id)
     
     Stripe::PermissionError:
       The provided key 'sk_test_******************************uCJm' does not have access to account 'acc_123' (or that account does not exist). Application access may have been revoked.
```
I feel like we might need to create a payment intent linked to a test customer, maybe adding `customer:` here (see [doc](https://stripe.com/docs/api/payment_intents/create)) : 
https://github.com/openfoodfoundation/openfoodnetwork/blob/4dda4e2385fcf45b95f4c2bfe745b2a98b7d42af/spec/models/spree/gateway/stripe_sca_spec.rb#L42-L49

What do you think ? Do you have a test customer I could use on your stripe test account ?

Reply:
Thanks for pinging me here @rioug . I think:
- we need to create customers by demand (with real Stripe calls), so that we don't use hard coded customer IDs.
- the error message points to `The provided key 'sk_test_******************************uCJm' does not have access to account 'acc_123'` because we're setting it with this value a few lines above. I think we need to create connected accounts as well, when needed (as above, with real Stripe calls), and then use the ID on line 90 above

I've spotted an inconsistency in a related spec, and try to fix it [here](https://github.com/openfoodfoundation/openfoodnetwork/pull/12127/files#). I think the setup could be used in this spec as well.

I hope this helps. What do you think?

Reply:
Thanks @filipefurtad0 I made progress on this but not quite there yet, and as David pointed out I can use my own stripe test account to get the test working, it should unblock me.